openmdao/error_checking/check_config.py

```diff
@@ -172,9 +176,16 @@ def _get_used_before_calc_subs(group, input_srcs):
             oparts = src_abs.split('.')
             src_sys = oparts[glen]
             tgt_sys = iparts[glen]
+
+            hierarchy_check = True
+            for i in top_level_hierarchy:
+                if i in tgt_abs:
+                    hierarchy_check = False
+
             if (src_sys in parallel_solver and tgt_sys in parallel_solver and
                     (parallel_solver[src_sys] not in ["NL: NLBJ", "NL: Newton", "BROYDEN"]) and
-                    src_sys == tgt_sys):
+                    src_sys == tgt_sys and
+                    hierarchy_check):
```

Comment:
I'm not sure, but I think this whole approach might be overkill.  If you connect `par_sys.anything` to `par_sys.anything`, you want to warn about the solvers.  If you connect `par_sys.any_group.anything` to `par_sys.any_group.anything`, then you don't want to warn.  Can't you just figure out your depth by checking the length of oparts and iparts beyond glen?

Reply:
I guess i should clarify, 

Warn for these (if missing the required solver):
`a.par.b.c` -> `a.par.d.e`
`a.par.b.c.d.e` -> `a.par.g.j.k`

But don't warn for these:
`a.par.b.c` -> `a.par.b.d`
`a.par.b.c.d.e` -> `a.par.b.c.g.h`

So maybe the easiest way is to check if there are at least 2 more strings after `glen`, and then move on if oparts[glen+1] and iparts[glen+1] are the same.

Reply:
That's a better solution. Implemented